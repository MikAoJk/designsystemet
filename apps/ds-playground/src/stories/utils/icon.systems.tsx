import { SystemSVGPathsAndIcons } from '@skatteetaten/ds-icons';

import {
  findSVGPathElements,
  makeLabelsFromSVGPathElements,
} from './icon.utils';

export const SystemSVGPaths = findSVGPathElements(SystemSVGPathsAndIcons);
export const SystemIconLabels = makeLabelsFromSVGPathElements(SystemSVGPaths);
