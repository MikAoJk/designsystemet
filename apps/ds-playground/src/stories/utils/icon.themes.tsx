import { ThemeSVGPathsAndIcons } from '@skatteetaten/ds-icons';

import {
  findSVGPathElements,
  makeLabelsFromSVGPathElements,
} from './icon.utils';

export const ThemeSVGPaths = findSVGPathElements(ThemeSVGPathsAndIcons);
export const ThemeIconLabels = makeLabelsFromSVGPathElements(ThemeSVGPaths);
