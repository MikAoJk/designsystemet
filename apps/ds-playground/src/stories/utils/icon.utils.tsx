import {
  SystemSVGPathsAndIcons,
  ThemeSVGPathsAndIcons,
} from '@skatteetaten/ds-icons';

export type SVGPathAndIconType =
  | typeof SystemSVGPathsAndIcons
  | typeof ThemeSVGPathsAndIcons;
export type SVGPathAndIconKey = keyof SVGPathAndIconType;

export const makeLabelsFromSVGPathElements = function (
  SVGPathElements: Record<SVGPathAndIconKey, SVGPathAndIconType>
): Record<string, string> {
  const labels: Record<string, string> = {};
  Object.keys(SVGPathElements).forEach((name: string) => {
    labels[name] = name.replace('SVGpath', '');
  });

  return labels;
};

export const findSVGPathElements = function (
  SVGPathAndIconElements: Record<SVGPathAndIconKey, SVGPathAndIconType>
): Record<string, SVGPathAndIconType[SVGPathAndIconKey]> {
  const elements: Record<string, SVGPathAndIconType[SVGPathAndIconKey]> = {};
  Object.keys(SVGPathAndIconElements).forEach((elementName: string) => {
    if (elementName.endsWith('SVGpath')) {
      elements[elementName] =
        SVGPathAndIconElements[elementName as SVGPathAndIconKey];
    }
  });

  return elements;
};
