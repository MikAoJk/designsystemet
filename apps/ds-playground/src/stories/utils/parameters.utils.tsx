import { ArgsTable, Stories, PRIMARY_STORY } from '@storybook/addon-docs';
import type { Parameters } from '@storybook/csf';

type DesignParameters = {
  name: string;
  url: string;
};

export const getParameters = function (
  designParameters: DesignParameters[]
): Parameters {
  return {
    docs: {
      page: (): JSX.Element => {
        return (
          <>
            <Stories title={''} />
            <ArgsTable story={PRIMARY_STORY} />
          </>
        );
      },
    },
    design: designParameters.map((designParameter: DesignParameters) => {
      return {
        name: designParameter.name,
        type: 'figma',
        url: designParameter.url,
      };
    }),
    backgrounds: {
      values: [
        { name: 'Svart', value: '#000' },
        { name: 'Hvit', value: '#fff' },
      ],
    },
  };
};
