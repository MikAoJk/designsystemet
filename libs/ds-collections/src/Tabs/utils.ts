export const valueRegex = /^[\w:.-_]*$/;
