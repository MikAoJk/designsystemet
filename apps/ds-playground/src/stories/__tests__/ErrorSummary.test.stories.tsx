import { headingAsArr } from '@skatteetaten/ds-core-utils';
import { ErrorMessage, ErrorSummary } from '@skatteetaten/ds-forms';
import { Paragraph } from '@skatteetaten/ds-typography';
import { expect } from '@storybook/jest';
import { StoryFn, Meta, StoryObj } from '@storybook/react';
import { userEvent, waitFor, within } from '@storybook/testing-library';

import { loremIpsum } from './testUtils/storybook.testing.utils';
import { webComponent } from '../../../.storybook/webcomponent-decorator';

const meta = {
  component: ErrorSummary,
  title: 'Tester/ErrorSummary/ErrorSummary',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    children: {
      table: { disable: true },
      control: { type: null },
    },
    content: {
      table: { disable: true },
      control: { type: null },
    },
    shadowRootNode: {
      table: { disable: true },
    },
    showErrorSummary: {
      table: { disable: true },
    },
    title: {
      table: { disable: true },
    },
    titleAs: {
      options: [...headingAsArr],
      control: 'inline-radio',
      table: { disable: true },
    },
  },
} satisfies Meta<typeof ErrorSummary>;
export default meta;
type Story = StoryObj<typeof meta>;

const defaultChildren = [
  <ErrorSummary.Error key={'errorSummaryError_1'} referenceId={'id_1'}>
    {'Husk å fylle ut type varer'}
  </ErrorSummary.Error>,
  <ErrorSummary.Error key={'errorSummaryError_2'} referenceId={'id_2'}>
    {'Selger du varer og tjenester for egen regning?'}
  </ErrorSummary.Error>,
];

export const WithRef = {
  name: 'With Ref (FA1)',

  args: {
    ref: (instance: HTMLDivElement | null): void => {
      if (instance) {
        instance.id = 'dummyIdForwardedFromRef';
      }
    },
  },

  argTypes: {
    ref: { table: { disable: false } },
  },

  parameters: {
    imageSnapshot: { disable: true },
  },

  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const container = canvas.getAllByRole('generic')[1];
    await expect(container).toHaveAttribute('id', 'dummyIdForwardedFromRef');
  },
} satisfies Story;

export const WithAttributes = {
  name: 'With Attributes (FA2-5, B1)',

  args: {
    id: 'htmlId',
    className: 'dummyClassname',
    lang: 'nb',
    'data-testid': '123ID',
    showErrorSummary: true,
  },

  argTypes: {
    id: { table: { disable: false } },
    className: { table: { disable: false } },
    lang: { table: { disable: false } },
    'data-testid': { table: { disable: false } },
  },

  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const container = canvas.getAllByRole('generic')[1];
    const errorSummary = canvas.getAllByRole('generic')[2];
    await expect(container).toHaveAttribute('id', 'htmlId');
    await expect(errorSummary).toHaveClass('dummyClassname');
    await expect(container).toHaveAttribute('lang', 'nb');
    await expect(container).toHaveAttribute('data-testid', '123ID');
  },
} satisfies Story;

export const Defaults = {
  name: 'Default (B2, B3 delvis)',

  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const container = canvas.getAllByRole('generic')[1];
    await expect(container).toBeInTheDocument();
    await expect(container).toHaveAttribute('aria-live', 'assertive');
    await expect(container).toHaveAttribute('aria-atomic');
    await expect(container).toHaveAttribute('tabIndex', '-1');
    // eslint-disable-next-line testing-library/no-node-access
    const errorSummary = container.querySelector('div');
    await expect(errorSummary).not.toBeInTheDocument();
  },
} satisfies Story;

export const WithChildren = {
  name: 'With Children (A1, A2, A3 delvis)',

  args: {
    showErrorSummary: true,
    children: defaultChildren,
  },

  argTypes: {
    children: { table: { disable: false } },
  },

  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const heading = canvas.getByRole('heading', { level: 2 });
    await expect(heading).toBeInTheDocument();
  },
} satisfies Story;

export const WithContent = {
  name: 'With Content (A1, A5, B1)',

  args: {
    showErrorSummary: true,
    title: 'Inntektsmottaker inneholder 2 feil:',
    content: <Paragraph>{loremIpsum}</Paragraph>,
  },

  argTypes: {
    content: { table: { disable: false } },
    title: { table: { disable: false } },
  },

  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const icon = canvas.getByRole('img', { hidden: true });
    await expect(icon).toBeInTheDocument();
    await expect(icon).toHaveAttribute('aria-hidden', 'true');
  },
} satisfies Story;

export const WithChildrenAndContent = {
  name: 'With Children And Content (A1, A3 delvis, A5)',

  args: {
    showErrorSummary: true,
    children: defaultChildren,
    content: <Paragraph>{loremIpsum}</Paragraph>,
  },

  argTypes: {
    children: { table: { disable: false } },
    content: { table: { disable: false } },
  },
} satisfies Story;

export const WithChildrenAndContentMobile = {
  name: 'With Children And Content Mobile',

  args: {
    showErrorSummary: true,
    children: defaultChildren,
    content: <Paragraph>{loremIpsum}</Paragraph>,
  },

  argTypes: {
    children: { table: { disable: false } },
    content: { table: { disable: false } },
  },

  parameters: {
    viewport: {
      defaultViewport: '--mobile',
    },
  },
} satisfies Story;

export const WithChildrenAndContentBreakpointXs = {
  name: 'With Children And Content Breakpoint Xs',

  args: {
    showErrorSummary: true,
    children: defaultChildren,
    content: <Paragraph>{loremIpsum}</Paragraph>,
  },

  argTypes: {
    children: { table: { disable: false } },
    content: { table: { disable: false } },
  },

  parameters: {
    viewport: {
      defaultViewport: '--breakpoint-xs',
    },
  },
} satisfies Story;

export const WithTitleAs = {
  name: 'With TitleAs (A2)',

  args: {
    showErrorSummary: true,
    titleAs: 'h1',
    children: defaultChildren,
  },

  argTypes: {
    titleAs: { table: { disable: false } },
  },

  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const heading = canvas.getByRole('heading', { level: 1 });
    await expect(heading).toBeInTheDocument();
  },
} satisfies Story;

const TemplateWithInput: StoryFn<typeof ErrorSummary> = () => (
  <>
    {/* TODO FRONT-1279 erstattes med TextField når den er ferdig utviklet */}
    <label className={'block'} htmlFor={'input_aar'}>
      {'År'}
    </label>
    <input id={'input_aar'} type={'number'} value={1009} />
    <ErrorMessage className={'bottomSpacingXL'} showError>
      {'Inntekståret må være etter 2008'}
    </ErrorMessage>
    <ErrorSummary showErrorSummary>
      <ErrorSummary.Error referenceId={'input_aar'}>
        {'Inntekståret må være etter 2008'}
      </ErrorSummary.Error>
    </ErrorSummary>
  </>
);

export const WithInput = {
  render: TemplateWithInput,
  name: 'With Input (A4)',

  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    const link = canvas.getByRole('link');
    const input = canvas.getByLabelText('År');
    await userEvent.click(link);
    await waitFor(() => expect(input).toHaveFocus());
  },
} satisfies Story;

const TemplateWithShadowRootNode: StoryFn<typeof ErrorSummary> = () => {
  // eslint-disable-next-line testing-library/no-node-access
  const element = document.querySelector('errorsummary-customelement');
  const shadowRoot = element?.shadowRoot;
  return (
    <div data-test-block>
      {/* TODO FRONT-1279 erstattes med TextField når den er ferdig utviklet */}
      <label className={'block'} htmlFor={'input_aar'}>
        {'År'}
      </label>
      <input id={'input_aar'} type={'number'} value={1009} />
      <ErrorMessage className={'bottomSpacingXL'} showError>
        {'Inntekståret må være etter 2008'}
      </ErrorMessage>
      <ErrorSummary shadowRootNode={shadowRoot ?? undefined} showErrorSummary>
        <ErrorSummary.Error referenceId={'input_aar'}>
          {'Inntekståret må være etter 2008'}
        </ErrorSummary.Error>
      </ErrorSummary>
    </div>
  );
};

export const WithShadowRootNode = {
  render: TemplateWithShadowRootNode,
  name: 'With ShadowRootNode',
  decorators: [webComponent],

  parameters: {
    a11y: {
      //Stilen som slår av transitions og animations finnes ikke på innsiden av shadowDom
      //Slår derfor av denne slik at vi ikke får false positives
      disable: true,
    },
    imageSnapshot: {
      disable: true,
    },
    customElementName: 'errorsummary-customelement',
  },

  play: async ({ canvasElement }): Promise<void> => {
    const canvas = within(canvasElement);
    // errorLink finnes ikke utenfor shadowDom
    await expect(canvas.queryByRole('link')).not.toBeInTheDocument();
    // eslint-disable-next-line testing-library/no-node-access
    const customElement = canvasElement.querySelector(
      `errorsummary-customelement`
    );
    await expect(customElement).toBeInTheDocument();
    const errorLink =
      // eslint-disable-next-line testing-library/no-node-access
      customElement?.shadowRoot && customElement.shadowRoot.querySelector('a');
    await expect(errorLink).toBeInTheDocument();
    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
    await userEvent.click(errorLink!);
    const input =
      customElement?.shadowRoot &&
      // eslint-disable-next-line testing-library/no-node-access
      customElement.shadowRoot.querySelector('input:focus');
    await expect(input).toBeInTheDocument();
  },
} satisfies Story;
