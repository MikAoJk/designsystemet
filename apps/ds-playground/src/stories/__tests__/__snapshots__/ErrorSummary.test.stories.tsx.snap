// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/ErrorSummary/ErrorSummary Defaults play-test 1`] = `
<div data-test-block="true">
  <div aria-live="assertive"
       tabindex="-1"
       aria-atomic="true"
  >
  </div>
</div>
`;

exports[`Tester/ErrorSummary/ErrorSummary WithAttributes play-test 1`] = `
<div data-test-block="true">
  <div id="htmlId"
       lang="nb"
       data-testid="123ID"
       aria-live="assertive"
       tabindex="-1"
       aria-atomic="true"
  >
    <div class="ErrorSummary_errorSummary__vciMR dummyClassname">
      <div class="ErrorSummary_iconContainer__Jhnlg">
        <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_large__c527o ErrorSummary_icon__vHubL"
             viewbox="0 0 24 24"
             focusable="false"
             aria-hidden="true"
             role="img"
        >
          <path d="M13 13h-2V7h2v6Zm0 4h-2v-2h2v2ZM12 2C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Z">
          </path>
        </svg>
      </div>
      <div class="ErrorSummary_summaryContainer__1LXQT">
        <h2 class="Heading_heading__7K0um Heading_heading_level4__KEMPA Heading_heading_hasSpacingSmall__ofrX+ ">
          For å gå videre må du rette opp 0 feil:
        </h2>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tester/ErrorSummary/ErrorSummary WithChildren play-test 1`] = `
<div data-test-block="true">
  <div aria-live="assertive"
       tabindex="-1"
       aria-atomic="true"
  >
    <div class="ErrorSummary_errorSummary__vciMR ">
      <div class="ErrorSummary_iconContainer__Jhnlg">
        <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_large__c527o ErrorSummary_icon__vHubL"
             viewbox="0 0 24 24"
             focusable="false"
             aria-hidden="true"
             role="img"
        >
          <path d="M13 13h-2V7h2v6Zm0 4h-2v-2h2v2ZM12 2C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Z">
          </path>
        </svg>
      </div>
      <div class="ErrorSummary_summaryContainer__1LXQT">
        <h2 class="Heading_heading__7K0um Heading_heading_level4__KEMPA Heading_heading_hasSpacingSmall__ofrX+ ">
          For å gå videre må du rette opp 2 feil:
        </h2>
        <ul class="LinkGroup_linkGroup__ZiEgs   ">
          <li class="LinkGroup_linkGroupItem__Y7SjF">
            <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_forward__ErPTi "
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M1.8 12.6v-1.8h16.8l-5.2-5.1 1.4-1.3 7.5 7.3-7.7 7.3-1.4-1.3 5.2-5.1H1.8Z">
              </path>
            </svg>
            <a class="Link_link__jeUpY  "
               href="#id_1"
            >
              Husk å fylle ut type varer
            </a>
          </li>
          <li class="LinkGroup_linkGroupItem__Y7SjF">
            <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_forward__ErPTi "
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M1.8 12.6v-1.8h16.8l-5.2-5.1 1.4-1.3 7.5 7.3-7.7 7.3-1.4-1.3 5.2-5.1H1.8Z">
              </path>
            </svg>
            <a class="Link_link__jeUpY  "
               href="#id_2"
            >
              Selger du varer og tjenester for egen regning?
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tester/ErrorSummary/ErrorSummary WithChildrenAndContent smoke-test 1`] = `
<div data-test-block="true">
  <div aria-live="assertive"
       tabindex="-1"
       aria-atomic="true"
  >
    <div class="ErrorSummary_errorSummary__vciMR ">
      <div class="ErrorSummary_iconContainer__Jhnlg">
        <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_large__c527o ErrorSummary_icon__vHubL"
             viewbox="0 0 24 24"
             focusable="false"
             aria-hidden="true"
             role="img"
        >
          <path d="M13 13h-2V7h2v6Zm0 4h-2v-2h2v2ZM12 2C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Z">
          </path>
        </svg>
      </div>
      <div class="ErrorSummary_summaryContainer__1LXQT">
        <h2 class="Heading_heading__7K0um Heading_heading_level4__KEMPA Heading_heading_hasSpacingSmall__ofrX+ ">
          For å gå videre må du rette opp 2 feil:
        </h2>
        <ul class="LinkGroup_linkGroup__ZiEgs   ">
          <li class="LinkGroup_linkGroupItem__Y7SjF">
            <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_forward__ErPTi "
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M1.8 12.6v-1.8h16.8l-5.2-5.1 1.4-1.3 7.5 7.3-7.7 7.3-1.4-1.3 5.2-5.1H1.8Z">
              </path>
            </svg>
            <a class="Link_link__jeUpY  "
               href="#id_1"
            >
              Husk å fylle ut type varer
            </a>
          </li>
          <li class="LinkGroup_linkGroupItem__Y7SjF">
            <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_forward__ErPTi "
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M1.8 12.6v-1.8h16.8l-5.2-5.1 1.4-1.3 7.5 7.3-7.7 7.3-1.4-1.3 5.2-5.1H1.8Z">
              </path>
            </svg>
            <a class="Link_link__jeUpY  "
               href="#id_2"
            >
              Selger du varer og tjenester for egen regning?
            </a>
          </li>
        </ul>
        <div class="ErrorSummary_content__jEDZS ">
          <p class="Paragraph_paragraph__NPzKO Paragraph_paragraph_standard__ObBkf  ">
            Lorem ipsum dolor sit amet. Alle som har laget en nettside, trengt litt fylltekst eller bare surfet rundt på nettet har antageligvis sett disse ordene, etterfulgt av en tilsynelatende eviglang tekst fylt med latinske liksomsetninger.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tester/ErrorSummary/ErrorSummary WithChildrenAndContentBreakpointXs smoke-test 1`] = `
<div data-test-block="true">
  <div aria-live="assertive"
       tabindex="-1"
       aria-atomic="true"
  >
    <div class="ErrorSummary_errorSummary__vciMR ">
      <div class="ErrorSummary_iconContainer__Jhnlg">
        <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_large__c527o ErrorSummary_icon__vHubL"
             viewbox="0 0 24 24"
             focusable="false"
             aria-hidden="true"
             role="img"
        >
          <path d="M13 13h-2V7h2v6Zm0 4h-2v-2h2v2ZM12 2C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Z">
          </path>
        </svg>
      </div>
      <div class="ErrorSummary_summaryContainer__1LXQT">
        <h2 class="Heading_heading__7K0um Heading_heading_level4__KEMPA Heading_heading_hasSpacingSmall__ofrX+ ">
          For å gå videre må du rette opp 2 feil:
        </h2>
        <ul class="LinkGroup_linkGroup__ZiEgs   ">
          <li class="LinkGroup_linkGroupItem__Y7SjF">
            <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_forward__ErPTi "
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M1.8 12.6v-1.8h16.8l-5.2-5.1 1.4-1.3 7.5 7.3-7.7 7.3-1.4-1.3 5.2-5.1H1.8Z">
              </path>
            </svg>
            <a class="Link_link__jeUpY  "
               href="#id_1"
            >
              Husk å fylle ut type varer
            </a>
          </li>
          <li class="LinkGroup_linkGroupItem__Y7SjF">
            <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_forward__ErPTi "
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M1.8 12.6v-1.8h16.8l-5.2-5.1 1.4-1.3 7.5 7.3-7.7 7.3-1.4-1.3 5.2-5.1H1.8Z">
              </path>
            </svg>
            <a class="Link_link__jeUpY  "
               href="#id_2"
            >
              Selger du varer og tjenester for egen regning?
            </a>
          </li>
        </ul>
        <div class="ErrorSummary_content__jEDZS ">
          <p class="Paragraph_paragraph__NPzKO Paragraph_paragraph_standard__ObBkf  ">
            Lorem ipsum dolor sit amet. Alle som har laget en nettside, trengt litt fylltekst eller bare surfet rundt på nettet har antageligvis sett disse ordene, etterfulgt av en tilsynelatende eviglang tekst fylt med latinske liksomsetninger.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tester/ErrorSummary/ErrorSummary WithChildrenAndContentMobile smoke-test 1`] = `
<div data-test-block="true">
  <div aria-live="assertive"
       tabindex="-1"
       aria-atomic="true"
  >
    <div class="ErrorSummary_errorSummary__vciMR ">
      <div class="ErrorSummary_iconContainer__Jhnlg">
        <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_large__c527o ErrorSummary_icon__vHubL"
             viewbox="0 0 24 24"
             focusable="false"
             aria-hidden="true"
             role="img"
        >
          <path d="M13 13h-2V7h2v6Zm0 4h-2v-2h2v2ZM12 2C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Z">
          </path>
        </svg>
      </div>
      <div class="ErrorSummary_summaryContainer__1LXQT">
        <h2 class="Heading_heading__7K0um Heading_heading_level4__KEMPA Heading_heading_hasSpacingSmall__ofrX+ ">
          For å gå videre må du rette opp 2 feil:
        </h2>
        <ul class="LinkGroup_linkGroup__ZiEgs   ">
          <li class="LinkGroup_linkGroupItem__Y7SjF">
            <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_forward__ErPTi "
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M1.8 12.6v-1.8h16.8l-5.2-5.1 1.4-1.3 7.5 7.3-7.7 7.3-1.4-1.3 5.2-5.1H1.8Z">
              </path>
            </svg>
            <a class="Link_link__jeUpY  "
               href="#id_1"
            >
              Husk å fylle ut type varer
            </a>
          </li>
          <li class="LinkGroup_linkGroupItem__Y7SjF">
            <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_forward__ErPTi "
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M1.8 12.6v-1.8h16.8l-5.2-5.1 1.4-1.3 7.5 7.3-7.7 7.3-1.4-1.3 5.2-5.1H1.8Z">
              </path>
            </svg>
            <a class="Link_link__jeUpY  "
               href="#id_2"
            >
              Selger du varer og tjenester for egen regning?
            </a>
          </li>
        </ul>
        <div class="ErrorSummary_content__jEDZS ">
          <p class="Paragraph_paragraph__NPzKO Paragraph_paragraph_standard__ObBkf  ">
            Lorem ipsum dolor sit amet. Alle som har laget en nettside, trengt litt fylltekst eller bare surfet rundt på nettet har antageligvis sett disse ordene, etterfulgt av en tilsynelatende eviglang tekst fylt med latinske liksomsetninger.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tester/ErrorSummary/ErrorSummary WithContent play-test 1`] = `
<div data-test-block="true">
  <div aria-live="assertive"
       tabindex="-1"
       aria-atomic="true"
  >
    <div class="ErrorSummary_errorSummary__vciMR ">
      <div class="ErrorSummary_iconContainer__Jhnlg">
        <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_large__c527o ErrorSummary_icon__vHubL"
             viewbox="0 0 24 24"
             focusable="false"
             aria-hidden="true"
             role="img"
        >
          <path d="M13 13h-2V7h2v6Zm0 4h-2v-2h2v2ZM12 2C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Z">
          </path>
        </svg>
      </div>
      <div class="ErrorSummary_summaryContainer__1LXQT">
        <h2 class="Heading_heading__7K0um Heading_heading_level4__KEMPA Heading_heading_hasSpacingSmall__ofrX+ ">
          Inntektsmottaker inneholder 2 feil:
        </h2>
        <div class="ErrorSummary_content__jEDZS ErrorSummary_contentWithoutChildren__mZUdF">
          <p class="Paragraph_paragraph__NPzKO Paragraph_paragraph_standard__ObBkf  ">
            Lorem ipsum dolor sit amet. Alle som har laget en nettside, trengt litt fylltekst eller bare surfet rundt på nettet har antageligvis sett disse ordene, etterfulgt av en tilsynelatende eviglang tekst fylt med latinske liksomsetninger.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tester/ErrorSummary/ErrorSummary WithInput play-test 1`] = `
<div data-test-block="true">
  <div class="bottomSpacingXL">
    <label for="input_aar"
           class="TextField_label__K0isy  undefined"
    >
      <span class>
        År
      </span>
    </label>
    <input id="input_aar"
           class="TextField_textbox__qbZlk    undefined"
           required
           aria-describedby="textFieldErrorId-:rh:"
           aria-invalid="true"
           value="1009"
    >
    <div id="textFieldErrorId-:rh:"
         aria-live="assertive"
         aria-atomic="true"
    >
      <div class="ErrorMessage_errorMessage__508oW ">
        <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_small__Yzxmo ErrorMessage_errorMessageIcon__2N-Qa"
             viewbox="0 0 24 24"
             focusable="false"
             aria-hidden="true"
             role="img"
        >
          <path d="M13 13h-2V7h2v6Zm0 4h-2v-2h2v2ZM12 2C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Z">
          </path>
        </svg>
        <p class="ErrorMessage_errorMessageText__e27TU">
          Inntekståret må være etter 2008
        </p>
      </div>
    </div>
  </div>
  <div aria-live="assertive"
       tabindex="-1"
       aria-atomic="true"
  >
    <div class="ErrorSummary_errorSummary__vciMR ">
      <div class="ErrorSummary_iconContainer__Jhnlg">
        <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_large__c527o ErrorSummary_icon__vHubL"
             viewbox="0 0 24 24"
             focusable="false"
             aria-hidden="true"
             role="img"
        >
          <path d="M13 13h-2V7h2v6Zm0 4h-2v-2h2v2ZM12 2C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Z">
          </path>
        </svg>
      </div>
      <div class="ErrorSummary_summaryContainer__1LXQT">
        <h2 class="Heading_heading__7K0um Heading_heading_level4__KEMPA Heading_heading_hasSpacingSmall__ofrX+ ">
          For å gå videre må du rette opp 1 feil:
        </h2>
        <ul class="LinkGroup_linkGroup__ZiEgs   ">
          <li class="LinkGroup_linkGroupItem__Y7SjF">
            <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_forward__ErPTi "
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M1.8 12.6v-1.8h16.8l-5.2-5.1 1.4-1.3 7.5 7.3-7.7 7.3-1.4-1.3 5.2-5.1H1.8Z">
              </path>
            </svg>
            <a class="Link_link__jeUpY  "
               href="#input_aar"
            >
              Inntekståret må være etter 2008
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tester/ErrorSummary/ErrorSummary WithRef play-test 1`] = `
<div data-test-block="true">
  <div aria-live="assertive"
       tabindex="-1"
       aria-atomic="true"
       id="dummyIdForwardedFromRef"
  >
  </div>
</div>
`;

exports[`Tester/ErrorSummary/ErrorSummary WithShadowRootNode play-test 1`] = `
<div data-test-block="true">
  <errorsummary-customelement>
  </errorsummary-customelement>
</div>
`;

exports[`Tester/ErrorSummary/ErrorSummary WithTitleAs play-test 1`] = `
<div data-test-block="true">
  <div aria-live="assertive"
       tabindex="-1"
       aria-atomic="true"
  >
    <div class="ErrorSummary_errorSummary__vciMR ">
      <div class="ErrorSummary_iconContainer__Jhnlg">
        <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_large__c527o ErrorSummary_icon__vHubL"
             viewbox="0 0 24 24"
             focusable="false"
             aria-hidden="true"
             role="img"
        >
          <path d="M13 13h-2V7h2v6Zm0 4h-2v-2h2v2ZM12 2C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Z">
          </path>
        </svg>
      </div>
      <div class="ErrorSummary_summaryContainer__1LXQT">
        <h1 class="Heading_heading__7K0um Heading_heading_level4__KEMPA Heading_heading_hasSpacingSmall__ofrX+ ">
          For å gå videre må du rette opp 2 feil:
        </h1>
        <ul class="LinkGroup_linkGroup__ZiEgs   ">
          <li class="LinkGroup_linkGroupItem__Y7SjF">
            <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_forward__ErPTi "
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M1.8 12.6v-1.8h16.8l-5.2-5.1 1.4-1.3 7.5 7.3-7.7 7.3-1.4-1.3 5.2-5.1H1.8Z">
              </path>
            </svg>
            <a class="Link_link__jeUpY  "
               href="#id_1"
            >
              Husk å fylle ut type varer
            </a>
          </li>
          <li class="LinkGroup_linkGroupItem__Y7SjF">
            <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_medium__VmCFz LinkGroup_icon__cSIB- LinkGroup_icon_forward__ErPTi "
                 viewbox="0 0 24 24"
                 focusable="false"
                 aria-hidden="true"
                 role="img"
            >
              <path d="M1.8 12.6v-1.8h16.8l-5.2-5.1 1.4-1.3 7.5 7.3-7.7 7.3-1.4-1.3 5.2-5.1H1.8Z">
              </path>
            </svg>
            <a class="Link_link__jeUpY  "
               href="#id_2"
            >
              Selger du varer og tjenester for egen regning?
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
